controllers/src/main/scala/cromwell/pipeline/controller/utils/PathMatchers.scala

```diff
@@ -0,0 +1,20 @@
+package cromwell.pipeline.controller.utils
+
+import akka.http.scaladsl.server.PathMatcher1
+import akka.http.scaladsl.server.PathMatchers.Segment
+import cats.data.Validated
+import cromwell.pipeline.datastorage.dto
+import cromwell.pipeline.model.wrapper
+
+import java.nio.file.{ Path, Paths }
+
+object PathMatchers {
+  val ProjectId: PathMatcher1[dto.ProjectId] = Segment.map(dto.ProjectId(_))
+  val Path: PathMatcher1[Path] = Segment.map(Paths.get(_))
+  val RunId: PathMatcher1[wrapper.RunId] = Segment.flatMap {
+    wrapper.RunId.fromString(_) match {
+      case Validated.Valid(runId) => Some(runId)
+      case _                      => None
+    }
+  }
```

Comment:
```suggestion
  val RunId: PathMatcher1[wrapper.RunId] = Segment.flatMap(wrapper.RunId.fromString(_).toOption)
```
Please always check whether the required method exists before reimplementing it yourself.

Reply:
Opps, sorry. I forgot about that. I`ll fix it.

Reply:
Done.